### PR TITLE
Update heap setting documentation in light of machine dependent heap

### PR DESCRIPTION
**Pull request by mark-vieira**
Documentation updates to our heap size recommendations as a follow up to #65905.

### CODE REVIEW

File: docs/reference/setup/advanced-configuration.asciidoc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+[source,txt]
+------------------
+-Xms2g <1>
+-Xmx2g <2>
+------------------
+<1> Set the minimum heap size to 2g.
+<2> Set the maximum heap size to 2g.
+
+Using `jvm.options.d` is the preferred method for configuring the heap size for
+production deployments.
```
Comment (jrodewig) [inline docs/reference/setup/advanced-configuration.asciidoc:71]: Tried to streamline this a bit:
```suggestion
In production, we recommend using `jvm.options.d` to configure heap sizes.
```

File: docs/reference/setup/advanced-configuration.asciidoc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+It is also possible to set the heap size via the `ES_JAVA_OPTS` environment
+variable. This is generally discouraged for production deployments but is useful
+for testing because it overrides all other means of setting JVM options.
```
Comment (jrodewig) [inline docs/reference/setup/advanced-configuration.asciidoc:75]: Tried to streamline this. I'd combine with the previous para.
```suggestion
For testing, you can also set the heap sizes using the `ES_JAVA_OPTS`
environment variable. The `ES_JAVA_OPTS` variable overrides all other JVM
options. We do not recommend using `ES_JAVA_OPTS` in production.
```

File: docs/reference/setup/advanced-configuration.asciidoc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ES_JAVA_OPTS="-Xms4000m -Xmx4000m" ./bin/elasticsearch <2>
+------------------
+<1> Set the minimum and maximum heap size to 2 GB.
+<2> Set the minimum and maximum heap size to 4000 MB.
+
+NOTE: Configuring the heap for the <<windows-service,Windows service>> is
+different than the above. The values initially populated for the Windows
+service can be configured as above but are different after the service has been
+installed. Consult the <<windows-service,Windows service documentation>> for
+additional details.
```
Comment (jrodewig) [inline docs/reference/setup/advanced-configuration.asciidoc:89]: Simplified the xref a little.
```suggestion
installed. See <<windows-service>>.
```

**Review by jrodewig — APPROVED** [review]
This looks good to me overall. I left a few comments and suggestions but nothing blocking.

I think creating the `Advanced configuration settings` page is a good move. However, we probably want to tell users that they shouldn't mess with these unless they know what they're doing. We get a lot of search traffic, and I could see someone landing here without other context.

It may also be worth auditing some of our xrefs to the current heap size settings. I found one during the review, but there may be others. I can do that as a separate effort if wanted though.

Thanks @mark-vieira!

File: docs/reference/setup/important-settings/heap-size.asciidoc [hunk 4]
```diff
@@ -2,86 +2,10 @@
 [discrete]
 === Heap size settings
 
-By default, {es} tells the JVM to use a heap with a minimum and maximum
-size of 1 GB. When moving to production, it is important to configure heap size
-to ensure that {es} has enough heap available.
+By default, {es} will auto-determine the appropriate heap size based on total system memory,
+and the set of roles assigned to the node. It is generally recommended that heap size determination
+be deferred to {es}. Should you have unique requirements that necessitate setting JVM heap size
+manually, refer to <<setting-jvm-heap-size,setting JVM heap size>> for additional guidance.
```
[the remark this answers is not in the document]
Comment (mark-vieira) [reply docs/reference/setup/important-settings/heap-size.asciidoc]: These are "generally" the same. I've standardized on "total system memory" here and also added a note clairfying what this means in a containerized deployment.

File: docs/reference/setup/advanced-configuration.asciidoc [hunk 5]
```diff
@@ -0,0 +1,89 @@
+[[advanced-configuration]]
+=== Advanced configuration settings
+
```
[the remark this answers is not in the document]
Comment (mark-vieira) [reply docs/reference/setup/advanced-configuration.asciidoc:3]: I've added some language here.

[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/setup/advanced-configuration.asciidoc:3]: New copy looks great to me! Thanks!

**Comment by mark-vieira** [comment]
Thanks for the review @jrodewig! I've applied your suggested and made a couple other minor tweaks.

**Pull request merged**